commitment/taproot.go

```diff
@@ -150,11 +178,33 @@ func (t *TapscriptPreimage) TapHash() (*chainhash.Hash, error) {
 // VerifyNoCommitment verifies that the preimage is not a Taproot Asset
 // commitment.
 func (t *TapscriptPreimage) VerifyNoCommitment() error {
-	if IsTaprootAssetCommitmentScript(t.SiblingPreimage) {
-		return ErrPreimageIsTapCommitment
-	}
+	switch {
+	// A preimage smaller than a valid Taproot Asset commitment script needs
+	// no further inspection.
+	case len(t.SiblingPreimage) < TaprootAssetCommitmentScriptSize:
+		return nil
+
+	case len(t.SiblingPreimage) == TaprootAssetCommitmentScriptSize:
+		if IsTaprootAssetCommitmentScript(t.SiblingPreimage) {
+			return ErrPreimageIsTapCommitment
+		}
 
-	return nil
+		return nil
+
+	default:
+		// The sibling may be an encoded tapleaf; remove the version
+		// prefix and inspect the script.
+		_, script, err := ScriptFromLeafPreimage(t.SiblingPreimage)
+		if err != nil {
+			return err
+		}
+
+		if IsTaprootAssetCommitmentScript(script) {
```

Comment:
Reminder that this func also needs to be updated. Then we should write a negative tests where we try to include a TAP v1 commitment as a sibling hash for the addr creation.

Reply:
The update would be to tie this version check to defined `asset.Version` s right? I don't recall any other changes needed.

Reply:
This is the "v0 commitment secretly has a v1 commitment as a sibling" scenario. 

What I think we want to do is loosen the check here: https://github.com/lightninglabs/taproot-assets/blob/eb3ed937f3c92e0d6a7315d914274ad0d4186cef/commitment/tap.go#L316-L318

So it asserts length, then has the equality check left in.